internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -353,6 +348,70 @@ func applyProperties(p *processRunner, pid int) error {
 		}
 	}
 
+	if p.opts.Priority != 0 {
+		if err := syscall.Setpriority(syscall.PRIO_PROCESS, pid, p.opts.Priority); err != nil {
+			return fmt.Errorf("failed to set priority of process %s to %d: %w", p, p.opts.Priority, err)
+		}
+	}
+
+	ioPriority, ioPrioritySet := p.opts.IOPriority.Get()
+	if ioPrioritySet {
+		err := setIOPriority(p, pid, ioPriority)
+		if err != nil {
+			return err
+		}
+	}
+
+	schedulingPolicy, schedulingPolicySet := p.opts.SchedulingPolicy.Get()
```

Comment:
we can merge here and above for readability

```suggestion
	if schedulingPolicy, schedulingPolicySet := p.opts.SchedulingPolicy.Get(); schedulingPolicySet {
```